### PR TITLE
Integrating Fluent2 HUD into main.

### PR DESCRIPTION
**Pull request by rdeassis**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This change integrates the new Fluent2 Heads Up Display into the main branch.
The new version is implemented in SwiftUI and therefore is compatible for use in SwiftUI only environments.
Modifiers that de SwiftUI HUD depends on are also being added (**AnimationCompletionModifier** and **PresentationModifier**)

For context, the PRs that merged this code into the fluent2-tokens (formerly vnext-tokens) branch are referenced below:

- #951 
- #952 
- #960 

### Verification

- Tests exercising Demo App scenarios.
- Validation on consuming codebase currently underway (devmain).

| Light Mode                                       | Dark Mode                                      |
|----------------------------------------------|--------------------------------------------|
| HUD_UIKit_light | HUD_UIKit_dark |
| HUD_SwiftUI_light | HUD_SwiftUI_dark |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [x] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1034)

### CODE REVIEW

File: ios/FluentUI/HUD/HUDView.swift [hunk 1]
```diff
@@ -1,206 +0,0 @@
...
-    struct HUD {
-        static var activityIndicator: UIColor = .white
-        static var background = UIColor(light: gray900.withAlphaComponent(0.9), dark: gray700)
-        static var text = UIColor(light: textOnAccent, dark: textPrimary)
-    }
-}
-
-// MARK: - HUDView
-
-class HUDView: UIView {
```
[the remark this answers is not in the document]
Comment (huwilkes) [reply ios/FluentUI/HUD/HUDView.swift:29]: Ah, fair.

**Review by huwilkes — APPROVED** [review]

**Review by mischreiber — CHANGES_REQUESTED** [review]
Blocking merge until we can figure out versioning.

**Comment by rdeassis** [comment]
> Blocking merge until we can figure out versioning.

This is not an API breaking change, so it shouldn't be tied to issues in versioning.

**Comment by rdeassis** [comment]
> > Blocking merge until we can figure out versioning.
> 
> This is not an API breaking change, so it shouldn't be tied to issues in versioning.

We will be release the HUD in the same release as other controls that will break the public API.
This will be released in the next minor version update (0.6.0) currently.

**Review by mischreiber — APPROVED** [review]

**Pull request merged**